collector/client.go

```diff
@@ -30,11 +31,19 @@ const (
 	requestProtocol = "GET cache_object://localhost/%s HTTP/1.0"
 )
 
+func encryptLoginPass(login string, password string) string {
+	if len(login)==0 {
+		return ""
+	} else {
+		return b64.StdEncoding.EncodeToString([]byte(login+":"+password))
+	}
+}
 /*NewCacheObjectClient initializes a new cache client */
-func NewCacheObjectClient(hostname string, port int) *CacheObjectClient {
+func NewCacheObjectClient(hostname string, port int, login string, password string) *CacheObjectClient {
 	return &CacheObjectClient{
 		hostname,
 		port,
+		encryptLoginPass(login, password),
```

Comment:
Please change the function name to `buildBasicAuthString` it's not really encrypting, just applying `base64`.